services/masterdata/src/lib/components/translation/translation.component.ts

```diff
@@ -0,0 +1,302 @@
+	private destroy$ = new Subject<void>();
+
+	isLoading = false;
+	languagesList: LanguageItem[] = [];
+	translationFlat: IMasterdataTranslationFlat[] = [];
+	translatableFields: string[] = [];
+	isDirtyChanges = false;
+
+	@Input() public showHeader = true;
+	@Input() public viewOnly = false;
+	@Input() public typeId = "";
+	@Input() public masterdataId = "";
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	@Input() public translation: IMasterdataTranslation<any>[] = [];
+
+	constructor(
+		private readonly dialogService: DialogService,
+		private readonly masterdataRenderer: MasterdataRendererService,
+		private readonly toastService: ToastService,
+		private readonly translateService: TranslateService,
+		private readonly service: MasterdataCrudHttpService
+	) {}
+
+	ngOnInit(): void {
+		this.initTranslation();
+		this.loadLanguagesList();
+	}
+
+	ngOnDestroy() {
+		this.destroy$.next();
+		this.destroy$.complete();
+	}
+
+	ngOnChanges() {
+		this.initTranslation();
+	}
+
+	isTypeModel(): boolean {
+		return !(this.masterdataId !== "");
+	}
+
+	loadLanguagesList() {
+		this.isLoading = true;
+		this.service.getAllLanguages(this.typeId, 0, 0).subscribe({
+			next: data => {
+				this.languagesList = data.value || [];
+				this.initTranslation();
+				this.isLoading = false;
+			},
+			complete: () => (this.isLoading = false),
+			error: () => (this.isLoading = false)
+		});
+	}
+
+	onSetDefaultLanguage(item: IMasterdataTranslationFlat) {
+		const { language } = item;
+		this.setDefaultLanguage(language);
+	}
+
+	onSaveChanges() {
+		this.isLoading = true;
+
+		if (this.isTypeModel()) {
+			const model = {} as IMasterdataTranslationUpdateMdType;
+			model.translations = this.mapFlatModel2Translation(this.translationFlat);
+			//console.log("SaveChange/model/type", model);
+
+			this.service.updateMasterdataTypeTranslation(this.typeId, model).subscribe({
+				next: data => {
+					this.saveChangesComplete(data, true);
+				},
+				complete: () => (this.isLoading = false),
+				error: () => (this.isLoading = false)
+			});
+		} else {
+			const model = {} as IMasterdataTranslationUpdateMdItem;
+			model.translations = this.mapFlatModel2Translation(this.translationFlat);
+			//console.log("SaveChange/model/masterdata", model);
+
+			this.service.updateMasterdataTranslation(this.typeId, this.masterdataId, model).subscribe({
+				next: data => {
+					this.saveChangesComplete(data, false);
+				},
+				complete: () => (this.isLoading = false),
+				error: () => (this.isLoading = false)
+			});
+		}
+	}
+
+	async onAddNewLanguage(): Promise<void> {
+		const translateKey = "addForm";
+		const headerText = this.translateService.instant(`masterdatamgmt.pages.masterdataTranslationsBox.upsert.${translateKey}.header`);
+		const dialogRef = this.dialogService.open(MasterdataTranslationUpsertComponent, {
+			header: headerText,
+			width: "50%",
+			data: {
```

Comment:
What about using a model here. Saves you also the parsing of the data object in MasterdataTranslationUpsertComponent a lot easier.

Reply:
Also for the upsert component